flixel/addons/editors/tiled/TiledMap.hx

```diff
@@ -45,24 +46,28 @@ class TiledMap
 	private var noLoadHash:Map<String, Bool> = new Map<String, Bool>();
 	private var layerMap:Map<String, TiledLayer> = new Map<String, TiledLayer>();
 	
-	private var rootPath:String="";
+	private var rootPath:String;
 	
 	/**
 	 * @param data Either a string or XML object containing the Tiled map data
 	 * @param rootPath Path to use as root to resolve any internal file references
 	 */
-	public function new(data:FlxTiledMapAsset, rootPath:String="")
+	public function new(data:FlxTiledMapAsset, ?rootPath:String)
 	{
 		var source:Fast = null;
 		
-		this.rootPath = rootPath;
+		if (rootPath != null)
+			this.rootPath = rootPath;
 		
 		if (Std.is(data, String)) 
 		{
+			if (this.rootPath == null)
+				this.rootPath = Path.directory(data) + "/";
 			source = new Fast(Xml.parse(Assets.getText(data)));
 		}
 		else if (Std.is(data, Xml)) 
 		{
+			this.rootPath = "";
```

Comment:
Hm.. is this correct? Maybe it should only be set if the `rootPath` parameter is `null`?

Reply:
I'll interpret the lack of a response as agreement. ;)